File: app/src/main/java/com/lostincontext/ruledetails/pick/BottomSheetGridItemViewHolder.java
package com.lostincontext.ruledetails.pick;


import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.lostincontext.commons.list.ViewHolder;
import com.lostincontext.databinding.ItemBottomSheetGridBinding;

public class BottomSheetGridItemViewHolder extends ViewHolder {


    private ItemBottomSheetGridBinding binding;

    public static BottomSheetGridItemViewHolder create(LayoutInflater inflater,
                                                       ViewGroup parent,
                                                       PickerDialogCallback callback) {
        ItemBottomSheetGridBinding itemBinding = ItemBottomSheetGridBinding.inflate(inflater,
                                                                                    parent,
                                                                                    false);
        return new BottomSheetGridItemViewHolder(itemBinding, callback);
    }


    public BottomSheetGridItemViewHolder(ItemBottomSheetGridBinding binding,
                                         PickerDialogCallback callback) {
        super(binding.getRoot());
        this.binding = binding;
        binding.setCallback(callback);
    }

    public void bindTo(GridBottomSheetItem item) {
        binding.setItem(item);
    }


}

File: app/src/main/java/com/lostincontext/ruledetails/pick/BottomSheetGridItemViewHolder.kt
package com.lostincontext.ruledetails.pick


import android.view.LayoutInflater
import android.view.ViewGroup

import com.lostincontext.commons.list.ViewHolder
import com.lostincontext.data.GridBottomSheetItem
import com.lostincontext.databinding.ItemBottomSheetGridBinding

class BottomSheetGridItemViewHolder(private val binding: ItemBottomSheetGridBinding,
                                    callback: PickerDialogCallback)
: ViewHolder(binding.root) {


    init {
        binding.callback = callback
    }

    fun bindTo(item: GridBottomSheetItem) {
        binding.item = item
    }

    companion object {

        fun create(inflater: LayoutInflater,
                   parent: ViewGroup,
                   callback: PickerDialogCallback): BottomSheetGridItemViewHolder {
            val itemBinding = ItemBottomSheetGridBinding.inflate(inflater,
                                                                 parent,
                                                                 false)
            return BottomSheetGridItemViewHolder(itemBinding, callback)
        }
    }


}
